src/ScriptCs.CSharp/CSharpScriptCompilerEngine.cs

```diff
@@ -2,33 +2,23 @@
 using System.IO;
 using System.Linq;
 using System.Reflection;
-using Roslyn.Scripting;
+using Microsoft.CodeAnalysis.Scripting;
+using Microsoft.CodeAnalysis.Scripting.CSharp;
 using ScriptCs.Contracts;
+using ScriptCs.Engine.Common;
 using ScriptCs.Exceptions;
 
-namespace ScriptCs.Engine.Roslyn
+namespace ScriptCs.CSharp
 {
-    public abstract class RoslynScriptCompilerEngine : RoslynScriptEngine
+    public abstract class CSharpScriptCompilerEngine : CommonScriptEngine
     {
         private const string CompiledScriptClass = "Submission#0";
 
         private const string CompiledScriptMethod = "<Factory>";
 
-        private readonly ILog _log;
-
-        [Obsolete("Support for Common.Logging types was deprecated in version 0.15.0 and will soon be removed.")]
-        protected RoslynScriptCompilerEngine(IScriptHostFactory scriptHostFactory, Common.Logging.ILog logger)
-            : this(scriptHostFactory, new CommonLoggingLogProvider(logger))
-        {
-        }
-
-        protected RoslynScriptCompilerEngine(IScriptHostFactory scriptHostFactory, ILogProvider logProvider)
+        protected CSharpScriptCompilerEngine(IScriptHostFactory scriptHostFactory, ILogProvider logProvider)
             : base(scriptHostFactory, logProvider)
         {
-            Guard.AgainstNullArgument("logProvider", logProvider);
-
-            _log = logProvider.ForCurrentType();
-
```

Comment:
I'm not a fan of this change. I don't think the base should expose it's `ILog`. The `ILogProvider` gets passed up the constructor call stack and each class can harvest it's own `ILog` from it. If derived classes use an `ILog` from a base class then the logger written when running under `-debug` will be the same for all the classes in the inheritance graph which is quite a loss of information.